src/MSIdentityScaffolding/Microsoft.DotNet.MSIdentity/MicrosoftIdentityPlatform/MicrosoftIdentityPlatformApplicationManager.cs

```diff
@@ -232,12 +241,12 @@ internal async Task<bool> UpdateApplication(TokenCredential tokenCredential, App
                     updatedApp.Web.ImplicitGrantSettings.EnableAccessTokenIssuance = toolOptions.EnableAccessToken.Value;
                 }
 
-                if (toolOptions.EnableIdToken.HasValue && (toolOptions.EnableIdToken.Value != updatedApp.Web.ImplicitGrantSettings.EnableIdTokenIssuance))
+                var idTokenEnabled = existingApplication.Web.ImplicitGrantSettings.EnableIdTokenIssuance ?? false;
+                if (!idTokenEnabled)
                 {
                     needsUpdate = true;
-                    updatedApp.Web.ImplicitGrantSettings.EnableIdTokenIssuance = toolOptions.EnableIdToken.Value;
+                    updatedApp.Web.ImplicitGrantSettings.EnableIdTokenIssuance = true;
```

Comment:
Do we know whether this is a VS tooling call to default it to true?

Reply:
In this case it sets the checkbox to true whenever VS calls into the tool to update the app registration, but there is no explicit parameter being sent from VS, is that the question?

Reply:
how are you determining if this is a VS call?

Reply:
Oh I see what you mean, let me fix that